### PR TITLE
Add IR utility that helps with return of multiple statements from Transform

### PR DESCRIPTION
**Pull request by vlstill**
Example of use is in the test: https://github.com/p4lang/p4c/pull/4414/files#diff-c05e1bbc752f9a1299cec11e3e39374bcd48de0eb2939bdca8b4fca6c0aa65d8R51.

### CODE REVIEW

File: test/gtest/transforms.cpp [hunk 1]
```diff
@@ -43,4 +44,38 @@ TEST_F(P4C_IR, Transform) {
...
+        }
+
+        const IR::Constant *c;
+    };
+
+    const auto *c2 = new IR::Constant(2);
+    const auto *c4 = new IR::Constant(4);
+    const auto *pe = new IR::PathExpression(new IR::Path("foo"));
+    const auto *stmt = new IR::BlockStatement(IR::IndexedVector<IR::StatOrDecl>(
+        {new IR::AssignmentStatement(pe, c2),
```
Comment (fruffy) [inline test/gtest/transforms.cpp:63]: You can get nicer formatting by using a trailing comma.

https://lists.llvm.org/pipermail/cfe-commits/Week-of-Mon-20140317/101748.html

File: ir/irutils.h [hunk 2]
```diff
@@ -108,6 +115,27 @@ big_int getMaxBvVal(int bitWidth);
 // This is 0 for unsigned and -(2^(t->size - 1)) for signed.
 big_int getMinBvVal(const Type *t);
 
+/// A helper useful when returning multiple statmenets out of a Transform's function that should
```
Comment (asl) [inline ir/irutils.h:118]: Typo, `statements`

File: test/gtest/transforms.cpp (hunk 1 above)
Comment (vlstill) [reply test/gtest/transforms.cpp]: Interesting, I did not even know that the trailing comma is accepted there.

File: ir/irutils.h [hunk 3]
```diff
@@ -108,6 +115,27 @@ big_int getMaxBvVal(int bitWidth);
 // This is 0 for unsigned and -(2^(t->size - 1)) for signed.
 big_int getMinBvVal(const Type *t);
 
+/// A helper useful when returning multiple statemenets out of a Transform's function that should
```
Comment (asl) [inline ir/irutils.h:118]: still a typo – `statemenets` :)

Comment (vlstill) [reply ir/irutils.h:118]: oh my, sorry

File: ir/irutils.cpp [hunk 4]
```diff
@@ -272,4 +273,36 @@ std::vector<const Expression *> flattenListOrStructExpression(const Expression *
...
 
+template <typename Stmts>
+const IR::Node *inlineBlockImpl(const Transform &t, Stmts &&stmts) {
+    if (stmts.size() == 1) {
+        // it could also be a declaration, and it that case, we need to wrap it in a block anyway
+        if (auto *stmt = (*stmts.begin())->template to<IR::Statement>()) {
+            return stmt;
+        }
+    }
+    if (t.getParent<const IR::Node>()->is<IR::BlockStatement>()) {
```
Comment (ChrisDodd) [inline ir/irutils.cpp:284]: could be just `if (t.getParent<IR::BlockStatement>())`

**Comment by ChrisDodd** [comment]
The way I've dealt with this in other passes is to just return a BlockStatement then use SimplifyControlFlow, but this is more efficient.

File: ir/irutils.cpp (hunk 4 above)
Comment (vlstill) [reply ir/irutils.cpp:284]: Right, for some reason I thought `getParent` fails if the type is not right, I probably confused it with `getOriginal`.

File: ir/irutils.cpp [hunk 5]
```diff
@@ -272,4 +273,36 @@ std::vector<const Expression *> flattenListOrStructExpression(const Expression *
...
+        // it could also be a declaration, and it that case, we need to wrap it in a block anyway
+        if (auto *stmt = (*stmts.begin())->template to<IR::Statement>()) {
+            return stmt;
+        }
+    }
+    if (t.getParent<IR::BlockStatement>()) {
+        return new IR::IndexedVector<IR::StatOrDecl>(std::forward<Stmts>(stmts));
+    }
+    Util::SourceInfo srcInfo;
+    if (stmts.size() > 0) {  // no .empty in initializer_list!
```
Comment (fruffy) [inline ir/irutils.cpp:288]: `std::empty`? I guess that won't work for the other two...

**Review by fruffy — APPROVED** [review]

File: ir/irutils.cpp (hunk 5 above)
Comment (vlstill) [reply ir/irutils.cpp:288]: It should work for anything that has .empty on c-style array and on initializer_list. But I could not decide if it is nicer or uglier than size() > 0.

**Pull request merged**